### DISCUSSION TITLE
How to read one agent data to another agent?

### DISCUSSION THREAD

**Discussion by JaslinRikkaz123** [mesa/mesa#1059] [Q&A]
Hi Mesa community...
I'm new to mesa software. I'm developing Electric Cars ABM. I need to use one agent function's return value to another agent. how to do that?
I have tried this, but it doesn't work:
        for agent in self.model.schedule.agents_dict:
            print(WeatherAgent.getOutdoorLight(self))
            
it gives :

  File "C:\Users\Jaslin Shaleemkhan\car2\solar\model.py", line 68, in __init__
    solarPanelAgent = SolarPanelAgent('Solar',self)
  File "C:\Users\Jaslin Shaleemkhan\car2\solar\agent.py", line 79, in __init__
    value.append(weatherlist)
NameError: name 'weatherlist' is not defined

Note: What i want is, temperature and irradiance data returned from weather agent it should be used in solar panel agent.

https://github.com/JaslinRikkaz123/EV_Model_inABM

**Comment by tpike3** [comment]
@JaslinRikkaz123  It looks like you fixed  this issues is that correct?

Regarding your other question: 
" I need to update model.py time(self.minute) parameter instead of steps in x-axis. How can I add it to my model? I am stuck at this moment. "

Unfortunately, it doesn't look like there is an easy fix. That capability in on the [TODO]( https://github.com/projectmesa/mesa/blob/72efecaacbc612927ff640f699f2cce4c01bc34e/mesa/visualization/modules/ChartVisualization.py#L36) 

I am looking for some quick work around but so far no luck.

(Sorry for the delay; new job at work)

Nice model by the way

**Comment by JaslinRikkaz123** [comment]
Thanks for your email
